reprounzip-singularity/Singularity_unpacker.py

```diff
@@ -0,0 +1,236 @@
+    cmd += "export SCIF_APPDATA_{0} SCIF_APPROOT_{1} SCIF_APPMETA_{2} SCIF_APPBIN_{3} SCIF_APPLIB_{4}\n".format(app_name,app_name,app_name,app_name,app_name)
+    cmd += "SCIF_APPENV_{0}=/scif/apps/{1}/scif/env/90-environment.sh\n".format(app_name,app_name)
+    cmd += "export SCIF_APPENV_{0}\n".format(app_name)
+    cmd += "SCIF_APPLABELS_{0}=/scif/apps/{1}/scif/labels.json\n".format(app_name,app_name)
+    cmd += "export SCIF_APPLABELS_{0}\n".format(app_name)
+    cmd += "SCIF_APPRUN_{0}=/scif/apps/{1}/scif/runscript\n".format(app_name,app_name)
+    cmd += "export SCIF_APPRUN_{0}\n".format(app_name)
+    with open(os.path.join(ENV_DIR, "94-appsbase.sh"), 'a') as f:
+        f.write(cmd)
+
+
+def make_app_specific_base_script(app_name):
+    cmd = "SCIF_APPNAME={}\n".format(app_name)
+    cmd += "SCIF_APPROOT=\"/scif/apps/{}\"\n".format(app_name)
+    cmd += "SCIF_APPMETA=\"/scif/apps/{}/scif\"\n".format(app_name)
+    cmd += "SCIF_DATA=\"/scif/data\"\n"
+    cmd += "SCIF_APPDATA=\"/scif/data/{}\"\n".format(app_name)
+    cmd += "SCIF_APPINPUT=\"/scif/data/{}/input\"\n".format(app_name)
+    cmd += "SCIF_APPOUTPUT=\"/scif/data/{}/output\"\n".format(app_name)
+    cmd += "export SCIF_APPDATA SCIF_APPNAME SCIF_APPROOT SCIF_APPMETA SCIF_APPINPUT SCIF_APPOUTPUT SCIF_DATA\n"
+    with open(os.path.join(APP_ENV_DIR, "01-base.sh"), 'w') as f:
+        f.write(cmd)
+
+# Check if bin is present in the tar if not add bin and sh
+def copy_busybox(tar):
+    if not "bin" in tar.getnames():
+        print("bin is absent")
+        tar.add("../../bin",arcname="bin")
+    elif not "sh" in tar.getnames():
+        print("bin but no sh")
+        tar.add("../../bin/sh", arcname='bin/sh')
+
+
+def deleteContent(fName):
+    with open(fName, "w"):
+        pass
+
+def add_singularity_folder(tar):
+    # check if the runs are multiple or single
+    my_dict = yaml.load(open(source_config))
+    runs = my_dict['runs']
+    copy_busybox(tar)
+    deleteContent(os.path.join(ENV_DIR, "94-appsbase.sh"))
+    if len(runs) > 1:
+	print("multiple runs")
+        # App structure is present
+        # First add the SCIF folder 
+        tar.add(SCIF_DIR, arcname="scif")
+        for run in runs:
+            binary = run['binary']
+            workingdir = run['workingdir']
+            run_file = run['argv'][1]
+            app_name = run['id']
+            make_runscript(workingdir, binary, run_file, apps=True)
+            write_env_file(runs[0].get('environ'), RUN_ENV_FILE, apps=True)
+            make_app_specific_base_script(app_name)
+            make_main_app_base_script(app_name)
+            # add the modified app.
+            tar.add(APP_DIR, arcname="scif/apps/" + app_name)
+
+    else:
+        write_env_file(runs[0].get('environ'),RUN_ENV_FILE)
+        make_runscript(runs[0]['workingdir'], runs[0]['binary'], runs[0]['argv'][1])
+    tar.add(SINGULARITY_DIR, arcname=".singularity.d")
```

Comment:
Is there a `.singularity.d/runscript` in the app case too? Does it run all the apps in order?